### PR TITLE
Add two universal field in fields.py

### PR DESCRIPTION
**Pull request by yingchaolu**
Added  MM_nuclei_density and El_number_density.

## PR Checklist

- [ ] Code passes flake8 checker
- [ ] New features are documented, with docstrings and narrative docs
- [ ] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

**Comment by yingchaolu** [comment]
I would also like to add universal fields for multiple temperature. But I did not find it in the naming convention.
http://ytep.readthedocs.io/en/latest/YTEPs/YTEP-0003.html#molecular-and-atomic-species-names

**Comment by ngoldbaum** [comment]
Can you explain a little bit more what multiple temperature means?

File: yt/frontends/flash/fields.py [hunk 1]
```diff
@@ -147,10 +147,14 @@ def _nele(field, data):
             return data["flash","dens"]*data["flash","ye"]*Na_code
         self.add_field(('flash','nele'), sampling_type="cell",  function=_nele, units="code_length**-3")
         self.add_field(('flash','edens'), sampling_type="cell",  function=_nele, units="code_length**-3")
+        self.add_field(('gas','El_number_density'), sampling_type="cell",  function=_nele,
+                       units=unit_system["number_density"])
         def _nion(field, data):
             Na_code = data.ds.quan(Na, '1/code_mass')
             return data["flash","dens"]*data["flash","sumy"]*Na_code
         self.add_field(('flash','nion'), sampling_type="cell",  function=_nion, units="code_length**-3")
+        self.add_field(('gas','MM_nuclei_density'), sampling_type="cell",  function=_nion,
```
Comment (ngoldbaum) [inline yt/frontends/flash/fields.py]: What does `MM` correspond to here?

Comment (yingchaolu) [reply yt/frontends/flash/fields.py]: Multi temperature means different species/component in the simulations have different temperatures, e.g ion/electron/radiation temperature in a 3T hydrodynamics, or electron/radiation temperature in a two fluid code.

Comment (ngoldbaum) [reply yt/frontends/flash/fields.py]: Oh I see, `MM` comes from a direct reading of YTEP-0003. In that document `MM` is a placeholder for the species name. We don't have a generic `MM_nuclei_density` field in yt. Instead, you should be using a specific species name. If `nion` doesn't correspond to a specific species, then just `ion_density` is probably a sufficiently universal name. Maybe @jzuhone or @brittonsmith could weigh in?

Comment (yingchaolu) [reply yt/frontends/flash/fields.py]: By MM, I mean the totality of atom/molecules (regardless of its ionization state and nuclei number Z)

Comment (ngoldbaum) [reply yt/frontends/flash/fields.py]: In that case I think just `ion_density` is the correct name. Like I mentioned in my comment, `MM` is not actually a shorthand we use inside yt, YTEP-0003 is just using it as a placeholder.

**Comment by ngoldbaum** [comment]
I don't think we thought about species temperature fields at all in YTEP-0003, so in principle we'd need to add to that system if we want to come up with a systematic way of naming these fields.

I think the field naming scheme that goes best with the naming scheme in YTEP-0003 is to associate temperature fields with each species name, e.g. `ion_temperature`, `El_temperature`, `C_temperature`, `radiation_temperature`, etc. Are there temperatures for individual ionization states or just for the species as a whole>

**Comment by yingchaolu** [comment]
I think in flash code we only need ion_temperature, El_temperature and radiation_temperature.

**Comment by ngoldbaum** [comment]
Hi, can you update this to fix the syntax error?

**Comment by matthewturk** [comment]
@yingchaolu  Anything I can do to help this get across the finish line?

**Comment by neutrinoceros** [comment]
I attempted to salvage this PR by merging from main here are my conclusions:
- the added aliases are no longer necessary
- the three temperature fields could still be added but maybe not universally (lots of fields are now conditionally defined in flash)
- we cannot push to the original branch because the fork has been deleted

Part of the patch may still be salvaged but I think the PR itself can't, so I'd recommend closing it. I can reopen as a different PR with @yingchaolu's commits.

**Comment by jzuhone** [comment]
I would simply open a new PR with the three temperature fields and that should be sufficient. But I am also not sure how relevant this is anymore. Either way I would say we can close it.

**Comment by neutrinoceros** [comment]
> I am also not sure how relevant this is anymore

@jzuhone ah that's too bad, I was hoping you'd have an idea !
Well, in the future if anyone want to pick this up again, please feel welcome to. In the mean time I will just close it.

**Comment by jzuhone** [comment]
I mean, I am pretty sure these fields are still used, but no one seems to be in a big hurry to implement them.

**Pull request closed without merging**